subworkflows/local/utils_nfcore_rnaseq_pipeline/main.nf

```diff
@@ -545,29 +527,79 @@ def biotypeInGtf(gtf_file, biotype) {
 }
 
 //
-// Function that parses and returns the predicted strandedness from the RSeQC infer_experiment.py output
+// Function to determine library type by comparing type counts. Consistent
+// between Salmon and RSeQC
 //
-def getInferexperimentStrandedness(inferexperiment_file, cutoff=30) {
-    def sense        = 0
-    def antisense    = 0
-    def undetermined = 0
+
+def calculateStrandedness(forwardFragments, reverseFragments, undeterminedFragments, threshold) {
+    def totalFragments = forwardFragments + reverseFragments + undeterminedFragments
+    def totalStrandedFragments = forwardFragments + reverseFragments
+
+    def strandedness = 'undetermined'
+    if (totalStrandedFragments > 0) {
+        def forwardProportion = forwardFragments / (totalStrandedFragments as double)
+        def reverseProportion = reverseFragments / (totalStrandedFragments as double)
+        def proportionDifference = Math.abs(forwardProportion - reverseProportion)
+
+        if (forwardProportion >= threshold) {
+            strandedness = 'forward'
+        } else if (reverseProportion >= threshold) {
+            strandedness = 'reverse'
```

Comment:
What happens if someone decide to put a threshold lower than `0.5`?

Reply:
I'd assume that in such a case all if statements here can be true, and then it's `forward` by default.